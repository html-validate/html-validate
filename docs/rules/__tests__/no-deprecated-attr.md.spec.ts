import HtmlValidate from '../../../src/htmlvalidate';

const markup = {};
markup["incorrect"] = `<body bgcolor="red"></body>`;
markup["correct"] = `<body style="background: red;"></body>`;

describe('docs/rules/no-deprecated-attr.md', () => {
	it('inline validation: incorrect', () => {
		const htmlvalidate = new HtmlValidate({"rules":{"no-deprecated-attr":"error"}});
		const report = htmlvalidate.validateString(markup["incorrect"]);
		expect(report.results).toMatchSnapshot();
	});
	it('inline validation: correct', () => {
		const htmlvalidate = new HtmlValidate({"rules":{"no-deprecated-attr":"error"}});
		const report = htmlvalidate.validateString(markup["correct"]);
		expect(report.results).toMatchSnapshot();
	});
});
