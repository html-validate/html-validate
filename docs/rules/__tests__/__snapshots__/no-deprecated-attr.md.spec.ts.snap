// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs/rules/no-deprecated-attr.md inline validation: correct 1`] = `Array []`;

exports[`docs/rules/no-deprecated-attr.md inline validation: incorrect 1`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "inline",
    "messages": Array [
      Object {
        "column": 7,
        "line": 1,
        "message": "Attribute \\"bgcolor\\" is deprecated on <body> element",
        "offset": 6,
        "ruleId": "no-deprecated-attr",
        "severity": 2,
        "size": 7,
      },
    ],
    "source": "<body bgcolor=\\"red\\"></body>",
    "warningCount": 0,
  },
]
`;
